api/src/main/java/io/smallrye/health/api/Startup.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+package io.smallrye.health.api;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Qualifier;
+
+import io.smallrye.common.annotation.Experimental;
+
+/**
+ *
+ * This qualifier is used to define a Startup Health Check procedure
+ *
+ * @author Martin Stefanko
+ */
+@Target({ TYPE, METHOD, PARAMETER, FIELD })
+@Retention(RUNTIME)
+@Documented
+@Qualifier
+@Experimental("Custom health check type definitions exposed at /health/startup. Not covered by the specification. " +
+        "Subject to change.")
+public @interface Startup {
```

Comment:
Not sure the name is the best choice. `@Startup` is already defined with EJBs for eager initialization. It might be used in other places as well.

I appreciate it's the name Kube used, but I'm concerned about creating confusion.

Reply:
Good point. Quarkus also has a `@Startup` annotation for eager initialization of CDI beans. `@StartupProbe` doesn't seem particularly attractive, as it raises a question "why not `@LivenessProbe`?"

Something like `@Started` perhaps? It doesn't play well with `@Liveness` / `@Readiness` / `@Wellness`, which is unfortunate.

Reply:
Maybe we can play off the definition in Kubernetes and use `@Initial`, or maybe `@SlowStartup`?

Reply:
Actually, I was also thinking about this, but I didn't come up with a better alternative. I am ok with all alternatives, maybe preferring a little `@SlowStartup` or maybe `@KubeStartup`. Really struggling with naming things.

Reply:
Maybe `@StartupProbe`, `@Start` or `@StartupHealth`?

Reply:
@ContainerStartup?

Reply:
I know this is horrible, but I'll say it anyway `@Startupness`

Reply:
**@StartupHealthCheck** ?

Reply:
Why not `@Startness` or `@Startupness`? Anyway,  `@Liveness` is not a word, so ...

Reply:
Liveness seems to be a word per wikipedia or Meriam-Webster :-) Funnily enough, I also found this: https://en.wiktionary.org/wiki/startness

Reply:
Dang, Cambridge Dictionary failed me! But if Wikipedia approves `startness`, we should definitely go for it ;)